pinot-spi/src/main/java/org/apache/pinot/spi/data/readers/Vector.java

```diff
@@ -0,0 +1,261 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.data.readers;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import java.util.Arrays;
+
+
+public class Vector implements Comparable<Vector> {
+  public enum VectorType {
+    FLOAT, INT
+  }
+
+  private int _dimension;
+  private float[] _floatValues;
+  private int[] _intValues;
+  private VectorType _type;
+
+  public Vector(int dimension, float[] values) {
+    _dimension = dimension;
+    _floatValues = values;
+    _type = VectorType.FLOAT;
+  }
+
+  public Vector(int dimension, int[] values) {
+    _dimension = dimension;
+    _intValues = values;
+    _type = VectorType.INT;
+  }
+
+  public int getDimension() {
+    return _dimension;
+  }
+
+  public void setDimension(int dimension) {
+    _dimension = dimension;
+  }
+
+  @JsonIgnore
+  public float[] getFloatValues() {
```

Comment:
[Lucene's HNSW index](https://github.com/apache/lucene/blob/dd4e66dad6726c53f2d89c5b7bcf74216949e4d3/lucene/core/src/java/org/apache/lucene/index/VectorEncoding.java#L40) supports only two kinds of types - float and bytes.

The Float type is same as the FLOAT type here. However, the byte type is for quantised float / int which is done mainly to compress size at the cost of precision. So if we end up having an index on the INT type here, we'd need a way to convert it to float[].

Just wanted to check your thoughts on this.

Reply:
Let's not mix vector and index.

Vector is just float/int array.

Index is based on different algorithms, distance measure and threshold etc, meaning one vector could also have multiple indexes for different use cases.

Reply:
Yes I agree. But I think we'd need support for quantised types here (in line with your comment below). So, if we add a BYTE type we should be good I think.

Reply:
Agreed. BYTE type should be supported here.